Adding DataLake* Swagger Specs

begoldsm opened the pull request:
This is the first round of DataLake\* swagger specs.
Known issues:
1. Variable host name in Job, Catalog and filesystem does not translate in swagger or code gen
2. Missing application/octet-stream as a valid content type for generation
3. No ability to define redirect behavior
4. No way to define an API that takes in a variable that is a full URI (when manually handling redirects)
5. No way to define that, if the return type is bytes, to return a byte array and not a string converted into a byte array (issue with code gen).

Commit:
Initial swagger json files for DataLake

Commit:
Remove extra layer parameters

Fix comments and remove extra layer "parameter" classes.
Put in proper return types for all classes.
Remove resource group and subscriptionId requirement for job and catalog
move the paths around for filesystem to work properly with swagger
uniqueness.

Commit:
minor update to have these actually be enums

Commit:
Updates to the swagger to have better naming

Swagger now supports naming the polymorphism properties with friendly
names and not based on the pivot type. Updating the swagger spec to use
better names.

azurecla commented:
Hi __@begoldsm__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (begoldsm). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

Commit:
removing an extra comma

Bad syntax is bad.

amarzavery commented:
@begoldsm The CI is failing. 
I think, it should be type object and format file for body parameter
{ 
46             "name": "streamContents", 
47             "in": "body", 
48             "schema": { 
49               "type": "file" 
50             }, 
51             "required": true, 
52             "description": "The file contents to include when appending to the file." 
53           },

Commit:
Fixing CI failure

switching to object and file instead of just file. Denis is working on a
separate PR to honor this properly in code gen.
